bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -102,12 +124,50 @@ public LinkyHandler(Thing thing, LocaleProvider localeProvider, Gson gson, HttpC
 
         this.cachedMonthlyData = new ExpiringDayCache<>("monthly cache", REFRESH_FIRST_HOUR_OF_DAY, () -> {
             LocalDate today = LocalDate.now();
-            return getConsumptionData(today.withDayOfMonth(1).minusMonths(1), today);
+            Consumption consumption = getConsumptionData(today.withDayOfMonth(1).minusMonths(1), today);
+            if (consumption != null) {
+                Aggregate days = consumption.aggregats.days;
+                Aggregate months = consumption.aggregats.months;
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Day {} value {}", days.periodes.get(days.periodes.size() - 1).dateDebut
+                            .format(DateTimeFormatter.ISO_LOCAL_DATE), days.datas.get(days.datas.size() - 1));
+                    for (int i = 0; i < months.datas.size(); i++) {
+                        logger.debug("Month {} {} value {}",
+                                months.periodes.get(i).dateDebut.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME),
+                                months.periodes.get(i).dateFin.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME),
+                                months.datas.get(i));
+                    }
+                }
+                if (days.datas.get(days.datas.size() - 1).isNaN()) {
+                    logger.debug("Monthly data including yesterday are not yet available");
+                    consumption = null;
+                }
```

Comment:
Near identical code is used in two other places. Please consider refactoring into another method and calling that instead.

Reply:
Done